setup.py

```diff
@@ -12,39 +12,48 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import print_function
 import os
 
-readme_filename = os.path.join(os.path.dirname(__file__), 'README.md')
-with open(readme_filename, 'r') as f:
-  readme = f.read()
+from setuptools import setup
+
+readme_filename = "README.md"
+current_directory = os.path.dirname(__file__)
+readme_path = os.path.join(current_directory, readme_filename)
 
 try:
-  import pypandoc
-  readme = pypandoc.convert(readme, to='rst', format='md')
-except:
-  print("Conversion of long_description from MD to reStructuredText failed...")
-  pass
+    with open(readme_path, 'r') as f:
+        readme = f.read()
+except Exception as e:
+    print (e)
+    print("Failed to open %s" % readme_path)
 
+try:
+    import pypandoc
+    readme = pypandoc.convert(readme, to='rst', format='md')
+except Exception as e:
```

Comment:
just chiming in from the sidelines: I'm always scared by super-braod `except` clauses in Python. They'll catch surprising things, e.g. `SyntaxError`.